### PR TITLE
Cache static resources, including plugin, theme, console, attachment

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind improvement
/area core
/milestone 2.1.x

#### What this PR does / why we need it:

Add cache control and last modified headers for responses of static resources.

#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/2966

#### Special notes for your reviewer:

We can check the response header of statis resources to know if the cache control is working correctly.

#### Does this PR introduce a user-facing change?

```release-note
完善静态资源的缓存策略
```

### CODE REVIEW

File: src/main/java/run/halo/app/config/WebFluxConfig.java [hunk 1]
```diff
@@ -119,16 +118,19 @@ private Mono<ServerResponse> redirectConsole(ServerRequest request) {
...
 
         // Mandatory resource mapping
         var uploadRegistration = registry.addResourceHandler("/upload/**")
             .addResourceLocations(FILE_URL_PREFIX + attachmentsRoot.resolve("upload") + "/")
             .setUseLastModified(true)
-            .setCacheControl(CacheControl.maxAge(Duration.ofDays(365)));
+            .setCacheControl(cacheControl);
 
         // For console project
         registry.addResourceHandler("/console/**")
```
Comment (ruibaby) [inline src/main/java/run/halo/app/config/WebFluxConfig.java:130]: 是否可以单独为 console/index.html 设置不缓存，因为 Halo 升级之后，index.html 中的资源文件都会改变，如果 index.html 有缓存，那么很可能在升级之后因为去请求旧的 js / css 资源导致无法正常访问。

Comment (JohnNiang) [reply src/main/java/run/halo/app/config/WebFluxConfig.java:130]: 参考：https://github.com/halo-dev/halo/blob/832f02b20e979b525760b07727809b2be0ae2ac0/src/main/java/run/halo/app/config/WebFluxConfig.java#L106-L108

Comment (ruibaby) [reply src/main/java/run/halo/app/config/WebFluxConfig.java:130]: nb

**Review by ruibaby — COMMENTED** [review]
/lgtm

image
image
image

**Review by guqing — COMMENTED** [review]
/approve

**Pull request merged**